components/integrations/bots/add_bot/index.ts

```diff
@@ -12,27 +12,31 @@ import {getUser} from 'mattermost-redux/selectors/entities/users';
 import {haveISystemPermission} from 'mattermost-redux/selectors/entities/roles';
 import {Permissions} from 'mattermost-redux/constants';
 
-import AddBot from './add_bot.jsx';
+import {GlobalState} from 'types/store';
 
-function mapStateToProps(state, ownProps) {
+import {ActionFunc} from 'mattermost-redux/types/actions';
+
+import AddBot, {Props} from './add_bot';
+
+function mapStateToProps(state: GlobalState, ownProps: Props) {
```

Comment:
Could you actually make a separate type for `OwnProps` containing the stuff that's used here and remove `location` from the component's `Props`? That's our usual pattern for doing this.

Reply:
Good to know for future. Done